tests/engines/registries/acceptance/branded/moderation/moderators-test.ts

```diff
@@ -63,4 +68,129 @@ module('Registries | Acceptance | branded.moderation | moderators', hooks => {
+        await click('[data-test-toggle-invite-form]');
+        await click('[data-test-confirm-add-moderator-button]');
+        await fillIn('[data-test-email-input]>div>input', 'testing@cos.io');
+        await fillIn('[data-test-full-name-input]>div>input', 'Baek Yerin');
+        await selectChoose('[data-test-select-permission]', 'Moderator');
+        await click('[data-test-confirm-add-moderator-button]');
+        assert.dom('#toast-container', document as any).hasTextContaining(
+            t(
+                'registries.moderation.moderators.addedNewModeratorSuccess',
+                { userName: 'Baek Yerin', permission: 'moderator' },
+            ),
+            'Toast successful message for inviting moderator by email',
+        );
+        assert.dom('[data-test-moderator-row]').exists({ count: 3 }, 'There are three moderators');
+        // Updating Suzy's permission
+        await clickTrigger(`[data-test-moderator-row="${suzy.id}"]`);
+        await selectChoose(`[data-test-moderator-row="${suzy.id}"]`, 'Admin');
+        assert.dom('#toast-container', document as any).hasTextContaining(
+            t(
+                'registries.moderation.moderators.updatedModeratorPermissionSuccess',
+                { userName: suzy.fullName, permission: 'admin' },
+            ),
+            'Toast successful message for updating permission',
+        );
+        assert.dom(`[data-test-moderator-row="${suzy.id}"]>div>[data-test-permission-group]`).hasText(
+            'Admin',
+            'Suzy is now an admin',
+        );
+        // Removing Suzy as a moderator
+        await click(`[data-test-delete-moderator-button="${suzy.id}"]>[data-test-delete-button]`);
+        await click('[data-test-confirm-delete]');
+        assert.dom('#toast-container', document as any).hasTextContaining(
+            t(
+                'registries.moderation.moderators.removedModeratorSuccess',
+                { userName: suzy.fullName },
+            ),
+            'Toast successful message for removing moderator',
+        );
+        assert.dom('[data-test-moderator-row]').exists({ count: 2 }, 'There are two moderators');
+        assert.dom(`[data-test-moderator-row="${suzy.id}"]`).doesNotExist('Suzy is no longer a moderator');
+    });
+
+    test('moderator CRUD operations failure should toast', async assert => {
+        server.namespace = '/v2';
+        server.post('/providers/registrations/:parentID/moderators', () => ({
+            errors: [{ detail: '' }],
+        }), 400);
+        server.put('/providers/registrations/:parentID/moderators/:id/', () => ({
+            errors: [{ detail: '' }],
+        }), 400);
+        server.patch('/providers/registrations/:parentID/moderators/:id/', () => ({
+            errors: [{ detail: '' }],
+        }), 400);
+        server.del('/providers/registrations/:parentID/moderators/:id/', () => ({
+            errors: [{ detail: '' }],
+        }), 400);
```

Comment:
Just wanted to comment on this for future reference. Good to overwrite mirage route handlers and request failures as well 👀